llvm/lib/CodeGen/GlobalISel/GIMatchTableExecutor.cpp

```diff
@@ -61,15 +61,41 @@ bool GIMatchTableExecutor::isBaseWithConstantOffset(
 
 bool GIMatchTableExecutor::isObviouslySafeToFold(MachineInstr &MI,
                                                  MachineInstr &IntoMI) const {
+  auto IntoMIIter = IntoMI.getIterator();
+
   // Immediate neighbours are already folded.
   if (MI.getParent() == IntoMI.getParent() &&
-      std::next(MI.getIterator()) == IntoMI.getIterator())
+      std::next(MI.getIterator()) == IntoMIIter)
     return true;
 
   // Convergent instructions cannot be moved in the CFG.
   if (MI.isConvergent() && MI.getParent() != IntoMI.getParent())
     return false;
 
-  return !MI.mayLoadOrStore() && !MI.mayRaiseFPException() &&
-         !MI.hasUnmodeledSideEffects() && MI.implicit_operands().empty();
+  if (MI.isLoadFoldBarrier())
```

Comment:
This PR has pulled in an entirely unrelated commit, any lowering patch shouldn't need to touch the selector tables

Reply:
Looks like https://github.com/llvm/llvm-project/pull/105689 accidentally got included here.

Reply:
I chose to include the TableGen patch to reduce the amount of regressions this patch creates to help with code review. This lowering patch was intended to help with selecting LD1 instructions but some of the patterns for LD1 were not matching due to reasons decribed in #101675.